src/EditorFeatures/CSharpTest/ImplementInterface/ImplementInterfaceTests.cs

```diff
@@ -8181,52 +8181,302 @@ public class Test : ITest
+
+    [return: NotNull]
+    T MethodDisallowNull([DisallowNull] T value);
+}}
+
+class Class : IInterface<string?>
+{{
+    public string? MethodAllowNull(string? value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+
+    [return: NotNull]
+    public string? MethodDisallowNull([DisallowNull] string? value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+}}");
+        }
+
+        // class Class<T> : IInterface<T?> where T : class
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestFlowAnalysisAttributeOnNonNullableReferenceTypeConstraint()
+        {
+            await TestInRegularAndScript1Async(
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
+    [return: MaybeNull]
+    T MethodAllowNull([AllowNull] T value);
+
+    [return: NotNull]
+    T MethodDisallowNull([DisallowNull] T value);
+}}
+
+class Class<T> : [|IInterface<T?>|] where T : class
+{{
+}}",
+$@"#nullable enable
 
 using System.Diagnostics.CodeAnalysis;
 
 {NullableAttributesCode}
 
-interface IFoo<T>
+interface IInterface<T>
 {{
+    [return: MaybeNull]
+    T MethodAllowNull([AllowNull] T value);
+
+    [return: NotNull]
+    T MethodDisallowNull([DisallowNull] T value);
+}}
+
+class Class<T> : IInterface<T?> where T : class
+{{
+    public T MethodAllowNull(T value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+
     [return: NotNull]
-    T Bar([DisallowNull] T bar);
+    public T MethodDisallowNull([DisallowNull] T value)
+    {{
+        throw new System.NotImplementedException();
+    }}
+}}");
+        }
+
+        // class Class<T> : IInterface<T> where T : class?
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestFlowAnalysisAttributeOnNullableReferenceTypeConstraint()
+        {
+            await TestInRegularAndScript1Async(
+$@"#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
 
+{NullableAttributesCode}
+
+interface IInterface<T>
+{{
     [return: MaybeNull]
-    T Baz([AllowNull] T bar);
+    T MethodAllowNull([AllowNull] T value);
+
+    [return: NotNull]
+    T MethodDisallowNull([DisallowNull] T value);
 }}
 
-class A : [|IFoo<int>|]
+class Class<T> : [|IInterface<T>|] where T : class?
 {{
 }}",
-@$"#nullable enable 
+$@"#nullable enable
 
 using System.Diagnostics.CodeAnalysis;
 
 {NullableAttributesCode}
 
-interface IFoo<T>
+interface IInterface<T>
 {{
+    [return: MaybeNull]
+    T MethodAllowNull([AllowNull] T value);
+
     [return: NotNull]
-    T Bar([DisallowNull] T bar);
+    T MethodDisallowNull([DisallowNull] T value);
+}}
+
+class Class<T> : IInterface<T> where T : class?
+{{
+    public T MethodAllowNull(T value)
+    {{
+        throw new System.NotImplementedException();
+    }}
 
+    public T MethodDisallowNull(T value)
```

Comment:
❕ Can't drop the attributes in this generated code

```suggestion
    [return: NotNull]
    public T MethodDisallowNull([DisallowNull] T value)
```

Reply:
How can I get the `?` in `class?`? I may need to specially handle the type parameters.

Reply:
The issue here is the behavior of `Class<object>` vs. `Class<object?>`. In both cases, the effective signature of this method is:

```
public object MethodDisallowNull(object value)
```

However, there is no way to represent this constraint in source without attributes.